pkg/webhook/validator.go

```diff
@@ -382,55 +384,56 @@ func ValidatePolicy(ctx context.Context, namespace string, ref name.Reference, c
-					result.err = errors.New("disallowed by static policy")
+					result.err = cosign.NewVerificationError("disallowed by static policy")
 					results <- result
 					return
 				}
-				result.signatures = []PolicySignature{{Subject: "allowed by static policy", Issuer: "allowed by static policy"}}
+
 			case len(authority.Attestations) > 0:
 				// We're doing the verify-attestations path, so validate (.att)
-				validatedAttestations, err := ValidatePolicyAttestationsForAuthority(ctx, ref, authority, authorityRemoteOpts...)
-				if err != nil {
-					result.err = err
-				} else {
-					result.attestations = validatedAttestations
-				}
+				result.attestations, result.err = ValidatePolicyAttestationsForAuthority(ctx, ref, authority, authorityRemoteOpts...)
+
 			default:
-				validatedSignatures, err := ValidatePolicySignaturesForAuthority(ctx, ref, authority, authorityRemoteOpts...)
-				if err != nil {
-					result.err = err
-				} else {
-					result.signatures = validatedSignatures
-				}
+				result.signatures, result.err = ValidatePolicySignaturesForAuthority(ctx, ref, authority, authorityRemoteOpts...)
 			}
 			results <- result
 		}()
 	}
+
 	// If none of the Authorities for a given policy pass the checks, gather
 	// the errors here. Even if there are errors, return the matched
 	// authoritypolicies.
-	authorityErrors := []error{}
+	authorityErrors := make([]error, 0, len(cip.Authorities))
 	// We collect all the successfully satisfied Authorities into this and
 	// return it.
-	policyResult := &PolicyResult{AuthorityMatches: make(map[string]AuthorityMatch)}
-	for i := 0; i < len(cip.Authorities); i++ {
+	policyResult := &PolicyResult{
+		AuthorityMatches: make(map[string]AuthorityMatch, len(cip.Authorities)),
+	}
+	for range cip.Authorities {
 		select {
 		case <-ctx.Done():
-			authorityErrors = append(authorityErrors, fmt.Errorf("context was canceled before validation completed"))
+			authorityErrors = append(authorityErrors, fmt.Errorf("%w before validation completed", ctx.Err()))
+
 		case result, ok := <-results:
 			if !ok {
-				authorityErrors = append(authorityErrors, fmt.Errorf("results channel failed to produce a result"))
+				authorityErrors = append(authorityErrors, errors.New("results channel closed before all results were sent"))
 				continue
 			}
 			switch {
 			case result.err != nil:
 				authorityErrors = append(authorityErrors, result.err)
+
 			case len(result.signatures) > 0:
 				policyResult.AuthorityMatches[result.name] = AuthorityMatch{Signatures: result.signatures}
+
 			case len(result.attestations) > 0:
 				policyResult.AuthorityMatches[result.name] = AuthorityMatch{Attestations: result.attestations}
+
 			default:
-				authorityErrors = append(authorityErrors, fmt.Errorf("failed to process authority: %s", result.name))
+				// This happens when we encounter a policy with:
+				//   static:
+				//     action: "pass"
+				policyResult.AuthorityMatches[result.name] = AuthorityMatch{}
```

Comment:
This is the `default` case I was referring to above.  We hit this when there wasn't an error, but there also weren't any validates `signatures` or `attestations`, which corresponds to the `static: { action: pass }` path.